### ISSUE TITLE
IMK Pinching Issue

### ISSUE THREAD

**Issue by jwberman** [amaelkady/Updated-IMK-Models-for-OpenSEES#2]
Hi Ahmed,

Thanks again for all your help getting the new IMK models running. I have a new problem with the IMK pinching model in the opensees you compiled for me a while back. (It is version 3.2.1 with a note that IMK is written by you, see screenshot below). The model parameters are listed below as well. Is this a known problem with IMKpinching?

![IMKPinchProblem](https://user-images.githubusercontent.com/93021970/174344087-ce7b71bf-3b93-444c-9c11-c5c7e220b6db.jpg)

![opensees version](https://user-images.githubusercontent.com/93021970/174355663-e0508fe3-7ed9-4211-89e8-dd36c91ac79e.jpg)

Parameters:

uniaxialMaterial IMKPinching 108 142.20000000 44.70000000 36.10000000 150.00000000 831.80000000 1.10000000 0.05000000 44.70000000 36.10000000 100.00000000 831.80000000 1.10000000 0.05000000 1000.00000000 1000.00000000 1000.00000000 1000.00000000 1.00000000 1.00000000 1.00000000 1.00000000 1.00000000 1.00000000 1.00000000 1.00000000;

Ke=142.2;
    Up_pos=44.7;
    Upc_pos=36.1;
    Uu_pos=150;
    Fy_pos=831.8;
    FmaxFy_pos=1.10;
    FresFy_pos=0.05;
    Up_neg=44.7;
    Upc_neg=36.1;
    Uu_neg=100;
    Fy_neg=831.8;
    FmaxFy_neg=1.10;
    FresFy_neg=0.05;
    Lamda_S=1000;
    Lamda_C=1000;
    Lamda_A=1000;
    Lamda_K=1000;
    c_S=1.0;
    c_C=1.0;
    c_A=1.0;
    c_K=1.0;
    D_pos=1;
    D_neg=1;
    kappa_F=1;
    kappa_D=1;

**Comment by amaelkady** [comment]
thank you for raising the issue.
Are you using the opensees executable provided in this repo?

**Comment by jwberman** [comment]
yep

**Comment by jwberman** [comment]
https://github.com/amaelkady/Updated-IMK-Models-for-OpenSEES/blob/master/OpenSees.exe

**Comment by jwberman** [comment]
This is Jeff Berman, we corresponded about 9 months ago and you put this version together for me, which is greatly appreciated!

**Comment by amaelkady** [comment]
ok; hi Jeff.
can you send me the deformation history from the figure above and I will check the model.

**Comment by jwberman** [comment]
disp = [5 10 15 20 30 40 50 60 70 80 95 100 110 120 130]; % input displacements for cyclic cycles
        Ncycles = [2 2 2 2 2 2 2 2 2 2 2 2 2 2 2]; % number of cycles per input displacement

**Comment by amaelkady** [comment]
Thanks but i need the exact deformation data points as extratced from opensees (the ones used in the plot)

**Comment by jwberman** [comment]
Oh, right. Sorry. 4th column in the attached.
[node_disp.txt](https://github.com/amaelkady/Updated-IMK-Models-for-OpenSEES/files/8933960/node_disp.txt)

**Comment by amaelkady** [comment]
thanks jeff. I will check it and let you know

**Comment by amaelkady** [comment]
Is there a reason why you are using the pinching model with kappa=1 instead of the peak-oriented model?

**Comment by jwberman** [comment]
You're right, i could use the peak oriented. We are using this because it
is a base case where will be adding pinching. I could switch to peak
oriented for kappa eqaul one if that is a problem then swith to pinching
for othe values. Wanted to try to stay consistent.

**Comment by amaelkady** [comment]
hi Jay, an updated opensees exe is uploaded now. check it and let me know if this issue is resolved.
